### ISSUE TITLE
Directly inlined iterators modify local tmp instead of arg

### ISSUE THREAD

**Issue by ronawho** [chapel-lang/chapel#7231]
### Summary of Problem
Updates to arguments passed to an iterator aren't preserved by direct iterator
inlining (--inline-iterators). e.g.

```chpl
iter myiter(in arg) {
  for i in 1..10 {
    arg += 1;
    yield arg;
  }
}

for i in myiter(1) do writeln(i);
```

will always yield `1` instead of `1, 2, 3, ...`

Note that this is a direct iterator inlining only bug. The test works if
inlining is done using the zippered iterator inlining code
(--no-inline-iterators --optimize-loop-iterators)

I _think_ this is likely a bug in replaceIteratorFormalsWithIteratorFields(),
but I'm not positive.

**Associated Future Test(s):**
[`test/functions/iterators/elliot/yieldArgIter/yieldArgIter.chpl`](https://github.com/chapel-lang/chapel/blob/master/test/functions/iterators/elliot/yieldArgIter/yieldArgIter.chpl) #7232

**Comment by ronawho** [comment]
I have a branch that attempts to inline all non-zippered iterators, not just those with only 1 yield, and ~12 tests break because of this bug.

Related to https://github.com/chapel-lang/chapel/issues/6757

**Comment by ronawho** [comment]
@daviditen Do you have any time to look at this?

**Comment by daviditen** [comment]
I looked at older versions to see if this has ever worked and 1.14.0 works.  I did a binary search to see when it broke and PR #5624 on March 16 is when it stopped working.  With that hint I'll continue looking.

**Comment by daviditen** [comment]
This change to [functionResolution.cpp:5585](https://github.com/chapel-lang/chapel/pull/5624/files#diff-e19fcde76769a1d3f15597da1da08352L5585) is where this test goes wrong.  Changing that line back makes the test pass.

The commit message for that [commit](https://github.com/chapel-lang/chapel/pull/5624/commits/eb60654d2e302cc8bdd4ba8560b430cf39ad70f6) states that the removed condition was apparently the reverse of what it should be.  I think this is correct, and the reverse should be added back in, along with another case for `fn->hasFlag(FLAG_ITERATOR_FN)` since we typically try to inline iterators.

I have this [change](https://github.com/chapel-lang/chapel/compare/master...daviditen:inlineIteratorBug?expand=1) in testing right now.

**Comment by daviditen** [comment]
[Test Summary - 170908.165338]
[Summary: #Successes = 7761 | #Failures = 0 | #Futures = 0]

**Comment by ronawho** [comment]
Closed by #7253
